website_sale_wishlist/models/product_wishlist.py

```diff
@@ -68,10 +68,12 @@ def _join_current_user_and_session(self):
         user_products = self.search([
             ("user_id", "=", self.env.uid),
         ]).mapped("product_tmpl_id")
-        session_domain = [
-            ("session", "=", request.session.sid),
-            ("user_id", "=", False),
-        ]
+
+        if request:
```

Comment:
It's also worth noting that this would still generate an error in the event of no session, because the domain is not defined in the no request case.